code/html/custom.js

```diff
@@ -287,23 +286,40 @@ function addValue(data, name, value) {
 
 function getData(form) {
 
+    // Populate two sets of data, ones that had been changed and ones that stayed the same
     var data = {};
+    var changed_data = {};
 
-    // Populate data
     $("input,select", form).each(function() {
         var name = $(this).attr("name");
+        if (name === "filename") { return; }
+        if (name === "rfbcode") { return; }
+
         var value = getValue(this);
         if (null !== value) {
+            if ("true" === $(this).attr("hasChanged")) {
+                addValue(changed_data, name, value);
+            }
             addValue(data, name, value);
         }
     });
 
-    // Post process
-    addValue(data, "schSwitch", 0xFF);
-    delete data["filename"];
-    delete data["rfbcode"];
+    // Finally, filter out only fields that had changed.
+    // Note: We need to preserve dynamic lists like schedules, wifi etc.
+    // so we don't accidentally break when user deletes entry in the middle
+    var resulting_data = {};
+    for (var value in data) {
+        if (value in changed_data) {
+            resulting_data[value] = data[value];
+        }
+    }
```

Comment:
> I would change the approach, get the entire data then do an object diff (non deep) between the original and that data, that's what I'm gonna do in vue

so this will handle the updated keys, if i understood the idea here.
group keys like scheduler will all break and be re-sent when, for example, one schedule is deleted.

not sure if this also could somehow handle completely deleted things, since there is no change there and they are simply gone from the tree